### PR TITLE
fix: [2298] `prefer-spread`: should not report on optional chaining

### PR DESCRIPTION
**Pull request by MichaelBlm**
Fixes https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2298

### CODE REVIEW

**Comment by MichaelBlm** [comment]
main branch fails `npm run integration -- --group 3 `

**Comment by fisker** [comment]
> @sindresorhus main branch fails `npm run integration -- --group 3 `

I'll fix that.

File: test/prefer-spread.mjs [hunk 1]
```diff
@@ -312,6 +312,8 @@ test.snapshot({
 		'buffer.slice()',
 		'file.slice()',
 		'class A {foo() {this.slice()}}',
+		'scopeManager?.scopes.slice()',
+		'(scopeManager?.scopes).slice()',
```
Comment (fisker) [inline test/prefer-spread.mjs:316]: I think we should report `(scopeManager?.scopes).slice()` since it's not an optional call.

Comment (MichaelBlm) [reply test/prefer-spread.mjs:316]: Ah it wasnt clear in the issue if the (scopeManager?.scopes).slice() was valid or invalid. Let me update the PR.

**Pull request merged**